test/es-module/test-typescript.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+const { spawnPromisified } = require('../common');
+const fixtures = require('../common/fixtures');
+const { match, strictEqual } = require('node:assert');
+const { test } = require('node:test');
+
+test('execute a typescript file', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    fixtures.path('typescript/test-typescript.ts'),
+  ]);
+
+  strictEqual(result.stderr, '');
+  match(result.stdout, /Hello, TypeScript!/);
+  strictEqual(result.code, 0);
+});
+
+test('execute a typescript file with imports', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    fixtures.path('typescript/b.ts'),
+  ]);
+
+  strictEqual(result.stderr, '');
+  match(result.stdout, /Hello, TypeScript!/);
+  strictEqual(result.code, 0);
+});
+
+test('execute a typescript with node_modules', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    fixtures.path('typescript/test-typescript-node-modules.ts'),
+  ]);
+
+  strictEqual(result.stderr, '');
+  match(result.stdout, /Hello, TypeScript!/);
+  strictEqual(result.code, 0);
+});
+
+test('expect error when executing a typescript file with imports with no extensions', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    fixtures.path('typescript/test-import-no-extension.ts'),
+  ]);
+
+  match(result.stderr, /Error \[ERR_MODULE_NOT_FOUND\]:/);
+  strictEqual(result.stdout, '');
+  strictEqual(result.code, 1);
+});
+
+test('expect error when executing a typescript file with enum', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    fixtures.path('typescript/test-enums.ts'),
+  ]);
+
+  // This error should be thrown during transformation
+  match(result.stderr, /TypeScript enum is not supported in strip-only mode/);
+  strictEqual(result.stdout, '');
+  strictEqual(result.code, 1);
+});
+
+test('expect error when executing a typescript file with experimental decorators', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    fixtures.path('typescript/test-experimental-decorators.ts'),
+  ]);
+  // This error should be thrown during transformation
+  match(result.stderr, /Decorators are not supported/);
```

Comment:
As ES decorators will become a part of JavaScript language (https://github.com/tc39/proposal-decorators),
the syntax, which is indistinguishable from TypeScript's legacy decorators, has led us to remove the associated errors.
In the upcoming release of swc, the presence of decorators will no longer trigger errors. See https://github.com/swc-project/swc/pull/9178.

Reply:
I thought the whole reason the legacy decorators were legacy was because the approved decorators proposal differed from the first decorators transform that TypeScript shipped? As in, `experimentalDecorators` is the old/non-standardized/legacy behavior, and the current unflagged syntax is what got standardized?

Reply:
The current state of decorators is quite similar to the situation with class fields. Please refer to this: https://github.com/nodejs/loaders/issues/208#issuecomment-2214003200

Reply:
If swc is not going to trasform decorators and leave them untouched, the javascript will be invalid because our v8 version does not support decorators afaik 🤔

Reply:
But V8 will support them eventually because they add all new syntaxes that become standardized. I think the relevance here is that we shouldn’t be testing for this, as some future V8 update will cause this test to start passing.

So if users write a decorator in their code, our SWC pass should leave it untouched, and then V8 will error on it (for now, until it starts working in a future version of V8). If there’s some way to test that our SWC output didn’t transform the decorator but left it as is, that would be what we’d want to test.